salt/cloud/clouds/ec2.py

```diff
@@ -3741,6 +3741,21 @@ def attach_volume(name=None, kwargs=None, instance_id=None, call=None):
                      provider=get_provider(),
                      opts=__opts__,
                      sigver='4')
+    while data[0]:
+        log.warn(
+            ('Error attaching volume {0} '
+            'to instance {1}. Retrying!').format(kwargs['volume_id'],
+                                                 instance_id))
+        # Instance isn't running yet, so cannot attach this volume to it
+        # wait for instance to run and try again
+        time.sleep(10)
```

Comment:
boto (100% sure on v3 not sure on v2) does have 

```
waiter = self._client.get_waiter('instance_running')
                waiter.wait(InstanceIds=[nat[0].id])
```
instead of using a hardcoded 10 sec value. My $0.02 cents